src/idl_gen_cpp.cpp

```diff
@@ -233,6 +233,28 @@ class CppGenerator : public BaseGenerator {
     return keywords_.find(name) == keywords_.end() ? name : name + "_";
   }
 
+  std::string Name(const FieldDef &field) const {
+    // the union type field suffix is immutable
+    static size_t union_suffix_len = strlen(UnionTypeFieldSuffix());
+    // early return if no case transformation required
```

Comment:
Is it though? The code still allows non-snake casing in the schema file, since it is just a warning.

Reply:
existing C++ code relies on the `_type()` suffix. The point of this PR is to negate the warning introduced by #6005 by migrating the schema, to snake-case to facilitate multi-language interoperability, without breaking C++ code that depends on the previous schema with camel-case.

Reply:
well there's the warning which continues to be emitted in that case

Reply:
basically I'm suggesting preserving current behavior with 100% fidelity, unless this option is set to something other than the default, in which case the transformation kicks in for C++ codebases that align to previous practice

Reply:
I see what you mean about fidelity, but by explicitly having the `--cpp-field-case` flag with `snake` as the option, would cause some confusion to users when it doesn't emit snake_case. They won't have the context to know why, at least now without this PR, we don't advertiser the casing.

Perhaps instead of having a `snake` option, you have a `schema-casing` or something similar, to better indicate what would happen.

Reply:
I guess "unchanged" or "passthru" is better, I'll do that

Reply:
done in latest commit